modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -153,6 +167,26 @@ public AmazonS3Reference client(RepositoryMetadata repositoryMetadata) {
         }
     }
 
+    /**
+     * Delegates to {@link #client(RepositoryMetadata)} when
+     * 1. per-project client is disabled
+     * 2. or when the blobstore is cluster level (projectId = null)
+     * Otherwise, attempts to retrieve a per-project client by the project-id and repository metadata from the
+     * per-project client manager. Throws if project-id or the client does not exist. The client maybe initialized lazily.
+     */
+    public AmazonS3Reference client(@Nullable ProjectId projectId, RepositoryMetadata repositoryMetadata) {
+        if (perProjectClientManager == null) {
+            // Multi-Project is disabled and we have a single default project
+            assert ProjectId.DEFAULT.equals(projectId) : projectId;
+            return client(repositoryMetadata);
+        } else if (projectId == null) {
+            // Multi-Project is enabled and we are retrieving a client for the cluster level blobstore
+            return client(repositoryMetadata);
+        } else {
+            return perProjectClientManager.client(projectId, repositoryMetadata);
+        }
+    }
```

Comment:
This is the main interface to access repo client for different scopes. Note the usage of `null` projectId for cluster level blobstore similar to how [cluster level persistent task](https://github.com/elastic/elasticsearch/blob/9b6ce866579435b7976f4434382781580855e881/server/src/main/java/org/elasticsearch/persistent/PersistentTasksClusterService.java#L153-L154) is handled.